odl/solvers/advanced/proximal_operators.py

```diff
@@ -258,90 +470,242 @@ def __init__(self, sigma):
+    # TODO: optimize
+    prox_cc_l2_squared = proximal_cconj_l2_squared(space, lam=lam, g=g)
+    return proximal_cconj(prox_cc_l2_squared)
+
+
+def proximal_cconj_l1(space, lam=1, g=None, isotropic=False):
+    """Proximal operator factory of the convex conjugate of the l1-norm.
+
+    Function for the proximal operator of the convex conjugate of the
+    functional F where F is an l1-norm
+
+        F(x) = lam ||x - g||_1
+
+    with x and g elements in ``space`` and scaling factor lam.
+
+    The convex conjugate, F^*, of F is given by the indicator function of
     the set box(lam)
 
-        F_cc(y) = lam ind_{box(lam)}(||y / lam||_p + <y / lam, g>)
+        F^*(y) = lam ind_{box(lam)}(|y / lam| + <y / lam, g>)
 
     where box(lam) is a hypercube centered at the origin with width 2 lam.
 
-    The proximal operator of F_cc is
+    The proximal operator of ``sigma * F^*`` where ``sigma`` is a step size
+    is given by
 
-        prox_sigma[F_cc](y) = lam (y - sigma g) / (max(lam 1_{||y||_p},
-        ||y - sigma g||_p)
+        prox[sigma * F^*](y) = lam (y - sigma g) / (max(lam, |y - sigma g|)
 
-    where max(.,.) thresholds the lower bound of ||y||_p point-wise and
-    1_{||y||_p} is a vector in the space of ||y||_p with all components set
+    An alternative formulation is available for `ProductSpace`'s, in that case
+    the ``isotropic`` parameter can be used, giving
+
+        F(x) = lam || ||x - g||_2 ||_1
+
+    In this case, the dual is
+
+        F^*(y) = lam ind_{box(lam)}(||y / lam||_2 + <y / lam, g>)
+
+    The proximal operator of ``sigma * F^*`` where ``sigma`` is a step size
+    is given by
+
+        prox[sigma * F^*](y) =
+            lam (y - sigma g) / (max(lam, ||y - sigma g||_2)
+
+    where max(.,.) thresholds the lower bound of ||y||_2 point-wise and
+    1 is a vector in the space of ||y||_2 with all components set
     to 1.
 
     Parameters
     ----------
-    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp` spaces
+    space : `LinearSpace` or `ProductSpace` of `LinearSpace` spaces
         Domain of the functional F
-    g : `DiscreteLpVector`
+    g : ``space`` element
         An element in ``space``
     lam : positive `float`
         Scaling factor or regularization parameter
+    isotropic : `bool`
+        True if the norm should first be taken pointwise. Only available if
+        ``space`` is a `ProductSpace`.
 
     Returns
     -------
-    prox : `Operator`
-        Returns the proximal operator to be initialized
+    prox_factory : `callable`
+        Factory for the proximal operator to be initialized
+
+    See Also
+    --------
+    proximal_l1 : proximal without convex conjugate conjugate
     """
     lam = float(lam)
 
-    if g is None:
-        g = space.zero()
-    else:
-        if g not in space:
-            raise TypeError('{!r} is not an element of {!r}'.format(g, space))
+    if g is not None and g not in space:
```

Comment:
That is not guaranteed to be copy-free, so thats why I didnt. Perhaps we should require copy-free-ness?